provider/maas/maas2_test.go

```diff
@@ -574,24 +574,124 @@ func (f *fakeFile) ReadAll() ([]byte, error) {
 }
 
 type fakeBlockDevice struct {
-	gomaasapi.BlockDevice
```

Comment:
You probably don't need to remove this embedded interface (along with the associated changes to implement all the methods on the fake struct). Embedding the interface means that the struct implements the interface without having to provide all the methods. It is ok to do that if the test doesn't call the methods directly (or else you'll get a nil pointer). If the test does call a method or two, you just need to implement those particular ones.

Reply:
So I originally did that and I was getting a panic segfault because my fakePartition was not implementing the method I actually called.

Just so I can be sure that the whole interface was implemented and I didn't want to worry about getting the segfault, I changed it to be this way.

It was more of a sanity thing, is that okay?

Reply:
The preference is not to add unneeded methods. We adopt everywhere else the interface embedding approach so please do that here also. You just need to implement the small number of methods actuallyused in order not to get a seg fault.

Reply:
I have also changed this back to only implement what is actually required.